pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -462,6 +621,139 @@ pub mod pallet {
 
 			Ok((address, incoming_msg))
 		}
+
+		/// Iterates over the outbound messages stored in the queue and attempts
+		/// to process them.
+		///
+		/// If a message fails to process it is moved to the
+		/// `FailedOutboundMessages` storage so that it can be executed again
+		/// via the `process_failed_outbound_message` extrinsic.
+		fn service_outbound_message_queue(max_weight: Weight) -> Weight {
+			let mut weight_used = Weight::zero();
+
+			let mut processed_entries = Vec::new();
+
+			for (nonce, (domain, sender, message)) in OutboundMessageQueue::<T>::iter() {
+				processed_entries.push(nonce);
+
+				let mut weight = Weight::zero();
+
+				match Self::process_message(domain.clone(), sender.clone(), message.clone()) {
+					Ok(post_info) => {
+						Self::deposit_event(Event::OutboundMessageExecutionSuccess {
+							nonce,
+							sender,
+							domain,
+							message,
+						});
+
+						weight = weight
+							.saturating_add(T::DbWeight::get().writes(1))
```

Comment:
CR: I suppose the `write` here refers to the removal of the entry. We also need to account for the read operation
```suggestion
							.saturating_add(T::DbWeight::get().reads_writes(1, 1))
```

Reply:
No, this `write` is for the event deposit operation. Any other weights should be added accordingly in `process_message`.

Reply:
I did add the missing 1 read tho, thanks for catching that!